kernel/kernel-api/src/test/scala/io/delta/kernel/internal/catalogManaged/CatalogManagedLogSegmentSuite.scala

```diff
@@ -0,0 +1,190 @@
+    testName = "Build RT with ratified commits that supersets the first checkpoint + deltas",
+    versionToLoad = 13L,
+    checkpointVersionOpt = Some(10L),
+    deltaVersions = 10L to 12L,
+    ratifiedCommitVersions = 9L to 13L,
+    expectedDeltaAndCommitVersionsOpt = Some(11L to 13L))
+
+  // _delta_log: [10.checkpoint+json, 11.json, 12.json                                 ]
+  // catalog:    [                             12.uuid.json, 13.uuid.json, 14.uuid.json]
+  testLogSegment(
+    testName = "Build RT with ratified commits that overlap with end of deltas",
+    versionToLoad = 14L,
+    checkpointVersionOpt = Some(10L),
+    deltaVersions = 10L to 12L,
+    ratifiedCommitVersions = 12L to 14L,
+    expectedDeltaAndCommitVersionsOpt = Some(11L to 14L))
+
+  // _delta_log: [10.checkpoint+json, 11.json, 12.json                           ]
+  // catalog:    [                                     13.uuid.json, 14.uuid.json]
+  testLogSegment(
+    testName = "Build RT with ratified commits that are after (no gap) the deltas",
+    versionToLoad = 14L,
+    checkpointVersionOpt = Some(10L),
+    deltaVersions = 10L to 12L,
+    ratifiedCommitVersions = 13L to 14L,
+    expectedDeltaAndCommitVersionsOpt = Some(11L to 14L))
+
+  // _delta_log: [0.json,                                      ]
+  // catalog:    [        1.uuid.json, 2.uuid.json, 3.uuid.json]
+  testLogSegment(
+    testName = "Build RT with only deltas and ratified commits",
+    versionToLoad = 3L,
+    checkpointVersionOpt = None,
+    deltaVersions = Seq(0L),
+    ratifiedCommitVersions = 1L to 3L,
+    expectedDeltaAndCommitVersionsOpt = Some(0L to 3L))
+
+  // _delta_log: [10.checkpoint+json,             ]
+  // catalog:    [                    11.uuid.json]
+  testLogSegment(
+    testName = "Build RT when checkpoint version is the last version from the filesystem",
+    versionToLoad = 11L,
+    checkpointVersionOpt = Some(10L),
+    deltaVersions = Seq(10L),
+    ratifiedCommitVersions = Seq(11L),
+    expectedDeltaAndCommitVersionsOpt = Some(Seq(11L)))
+
+  /////////////////////////////////////////////////////
+  // LogSegment construction tests -- negative cases //
+  /////////////////////////////////////////////////////
+
+  // _delta_log: [10.checkpoint+json, 11.json, 12.json                                ]
+  // catalog:    [                                          14.uuid.json, 15.uuid.json]
+  testLogSegment(
+    testName = "Build RT with ratified commits that are after (with gap) the deltas => ERROR",
+    versionToLoad = 15L,
+    checkpointVersionOpt = Some(10L),
+    deltaVersions = 10L to 12L,
+    ratifiedCommitVersions = 14L to 15L,
+    expectedExceptionClassOpt = Some(classOf[io.delta.kernel.exceptions.InvalidTableException]))
+
+  // _delta_log: [                                                  ]
+  // catalog:    [0.uuid.json, 1.uuid.json, 2.uuid.json, 3.uuid.json]
+  testLogSegment(
+    testName = "Build RT with only ratified commits => ERROR",
```

Comment:
Why is this an error? Because want to ensure that the "i'm catalog managed" metadata lands in the actual log?

Reply:
Great question -- This is something I / we need to think more deeply about:

The Engine and Catalog ultimately decide what it means to create a table.

If the "table" purely exists in the catalog -- in theory, then, you could point kernel at any empty path, and give it N ratified commits starting at 0 -- and say that's the table.

I'd need to think if this is a case we want to support

Reply:
I think it is -- What I will do for now is disable this test, and add a followup to support this

Basically -- it means we shouldn't LIST at all --> since all our LIST code tells us that an empty path is NOT a delta table

We will probably just check in SnapshotManager::getLogSegmentForVersion if the input list of LogDatas can be converted to a complete LogSegment --> and if so, just exit early